rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -693,6 +706,157 @@ void getOldFederationBtcUTXOs_calledTwice_returnCachedUtxos() {
+                assertArrayEquals(new byte[]{(byte) 0xaa}, data);
+                Assertions.assertEquals(networkParametersReceived, networkParameters);
+                return newFederation;
+            });
+
+            Assertions.assertEquals(newFederation, federationStorageProvider.getNewFederation(federationConstants, activationsBeforeFork));
+            Assertions.assertEquals(newFederation, federationStorageProvider.getNewFederation(federationConstants, activationsBeforeFork));
+            Assertions.assertEquals(2, storageCalls.size());
+            Assertions.assertEquals(1, deserializeCalls.size());
+        }
+    }
+
+    @Test
+    void getNewFederation_initialVersion_nullBytes() {
+        try (MockedStatic<BridgeSerializationUtils> bridgeSerializationUtilsMocked = mockStatic(BridgeSerializationUtils.class, CALLS_REAL_METHODS)) {
+            List<Integer> storageCalls = new ArrayList<>();
+            Repository repositoryMock = mock(Repository.class);
+            FederationStorageProvider federationStorageProvider = createFederationStorageProvider(repositoryMock);
+
+            when(repositoryMock.getStorageBytes(any(RskAddress.class), any(DataWord.class))).then((InvocationOnMock invocation) -> {
+                storageCalls.add(0);
+                RskAddress contractAddress = invocation.getArgument(0);
+                DataWord address = invocation.getArgument(1);
+
+                assertEquals(bridgeAddress, contractAddress);
+
+                if (storageCalls.size() == 1) {
+                    Assertions.assertEquals(NEW_FEDERATION_FORMAT_VERSION.getKey(), address);
+                    // First call is storage version getter
+                    return new byte[0];
+                } else {
+                    // Second and third calls are actual storage getters
+                    Assertions.assertTrue(storageCalls.size() == 2 || storageCalls.size() == 3);
+                    Assertions.assertEquals(NEW_FEDERATION_KEY.getKey(), address);
+                    return null;
+                }
+            });
+
+            assertNull(federationStorageProvider.getNewFederation(federationConstants, activationsBeforeFork));
+            assertNull(federationStorageProvider.getNewFederation(federationConstants, activationsBeforeFork));
+            Assertions.assertEquals(3, storageCalls.size());
+
+            bridgeSerializationUtilsMocked.verify(() -> BridgeSerializationUtils.deserializeStandardMultisigFederationOnlyBtcKeys(any(byte[].class), any(NetworkParameters.class)), never());
+            bridgeSerializationUtilsMocked.verify(() -> BridgeSerializationUtils.deserializeStandardMultisigFederation(any(byte[].class), any(NetworkParameters.class)), never());
+        }
+    }
+
+    @Test
+    void getNewFederation_multiKeyVersion() {
+        Federation newFederation = buildMockFederation(100, 200, 300);
+        testGetNewFederationPostMultiKey(newFederation);
```

Comment:
What do you think of parameterizing this test instead of using this common method between different tests?